### ISSUE TITLE
[Accessibility] Search sources widget needs to be accessible.

### ISSUE THREAD

**Issue by yzen** [devtools-html/debugger.html#4070]
Here is a list of accessibility improvements for the search sources interface (note: autocomplete accessibility should be worked on in #4071 ):

#### Search Input
- [x] Magnifying glass icon should have an ARIA role of "presentation" as it doesn't carry any meaningful semantics.
- [x] input element should have a type of "search".
- [x] input element should have keyboard focus styling.
- [ ] Results number section "X results" should have an ID set. This way we can apply an ARIA attribute [aria-describedby][by] on the input field.
- [x] Close button should be an actual button or at least have a role "button". It should have styling for focused-with-keyboard case. It should also be focusable and actionable with the keyboard.

#### Bottom Bar

- [ ] "Search sources..." and "Find in files…" are not accessible. Semantically they should be [radio][rb] buttons that are mutually exclusive. User should also be able to navigate to them using the keyboard only (tabbing to). They should also be actionable when Enter/Space are pressed.
- [ ] When "Search sources..." is selected and keyboard focus is inside the input there is a keyboard trap. There is no way to tab out of the input field (neither forward nor back).

### Screenshot

![Screen Shot 2017-09-18 at 8.08.43 PM.png](https://shipusercontent.com/42c1531e51e518e514c496914189cb4d/Screen%20Shot%202017-09-18%20at%208.08.43%20PM.png)

[by]: https://www.w3.org/TR/wai-aria/states_and_properties#aria-describedby
[rb]:https://www.w3.org/TR/wai-aria/roles#radio

**Comment by yzen** [comment]
### Magnifying glass icon should have an ARIA role of "presentation" as it doesn't carry any meaningful semantics.
This is an example of how to add an appropriate role to an element. In this case role "presentation" (https://www.w3.org/TR/wai-aria/roles#presentation) to an icon.
```diff
diff --git a/assets/images/Svg.js b/assets/images/Svg.js
index 956fe45b..ab67d81e 100644
--- a/assets/images/Svg.js
+++ b/assets/images/Svg.js
@@ -58,10 +58,11 @@ type SvgType = {
   name: string,
   clasName?: string,
   onClick?: () => void,
-  "aria-label"?: string
+  "aria-label"?: string,
+  role?: string,
 };
 
-function Svg({ name, className, onClick, "aria-label": ariaLabel }) {
+function Svg({ name, className, onClick, "aria-label": ariaLabel, role }) {
   if (!svg[name]) {
     const error = `Unknown SVG: ${name}`;
     if (isDevelopment()) {
@@ -80,6 +81,7 @@ function Svg({ name, className, onClick, "aria-label": ariaLabel }) {
     className,
     onClick,
     ["aria-label"]: ariaLabel,
+    role,
     src: svg[name]
   };
   return <InlineSVG {...props} />;
diff --git a/src/components/shared/SearchInput.js b/src/components/shared/SearchInput.js
index b8966fa1..2359ce6f 100644
--- a/src/components/shared/SearchInput.js
+++ b/src/components/shared/SearchInput.js
@@ -55,10 +55,10 @@ class SearchInput extends Component {
 
   renderSvg() {
     if (this.shouldShowErrorEmoji()) {
-      return <Svg name="sad-face" />;
+      return <Svg name="sad-face" role="presentation" />;
     }
 
-    return <Svg name="magnifying-glass" />;
+    return <Svg name="magnifying-glass" role="presentation" />;
   }
 
   renderArrowButtons() {
```

**Comment by yzen** [comment]
### input element should have a type of "search".
```diff
diff --git a/src/components/shared/SearchInput.js b/src/components/shared/SearchInput.js
index b8966fa1..3781cf6b 100644
--- a/src/components/shared/SearchInput.js
+++ b/src/components/shared/SearchInput.js
@@ -113,6 +113,7 @@ class SearchInput extends Component {
       className: classnames({
         empty: this.shouldShowErrorEmoji()
       }),
+      type: "search",
       onChange,
       onKeyDown,
       onKeyUp,
```

**Comment by tohmjudson** [comment]
I am going to get started on this one :)

**Comment by asharron** [comment]
Hello, I will be working to fix this issue along with @belld19233, @emmet0r, @josezindia, and @Jessonsotoventura

**Comment by asharron** [comment]
/claim


Hello, I will be working to fix this issue along with @belld19233, @emmet0r, @josezindia, and @Jessonsotoventura

**Comment by jasonLaster** [comment]
Welcome @asharron, @belld19233, @emmet0r, @josezindia, and @Jessonsotoventura!

**Comment by emmet0r** [comment]
Testing bot updates with /claim

**Comment by emmet0r** [comment]
Testing bot updates with /claim

**Comment by emmet0r** [comment]
/claim

I have some students who will be picking up where we left off with this ticket last term - will tag them shortly.

**Comment by mohamedabo** [comment]
Hello. We are students in @emmet0r class and we are claiming this issue. Comparing our own view of the debugger and this screenshot, it looks like both the "Search sources" and "Find in files" buttons have already been made keyboard accessible.
![image 1](https://user-images.githubusercontent.com/32283283/37672351-2706fba4-2c44-11e8-9d07-6aecf6cfc857.png)

![image](https://user-images.githubusercontent.com/32283283/37672330-1be7bd12-2c44-11e8-8964-242975b8719d.png)

**Comment by jasonLaster** [comment]
CC @yzen @darkwing

**Comment by Malachi-Holden** [comment]
/claim
Hello,
I'm on  a team with @BadInternetGuy, and @Rayje1997, in @emmet0r class, and we're working on the 5th checkbox, making the close button an actual button.

**Comment by Malachi-Holden** [comment]
/unclaim
I worked on this issue with @BadInternetGuy, and @Rayje1997, in @emmet0r class. We worked on the fifth checkbox. We got our PR merged, so we are now unclaiming the issue. Thanks to everyone who helped and gave us advice!

**Comment by tapiad** [comment]
Hello, 
I would like to help out. I am new to all this but would love to contribute. Would anyone help choose a bug I can start with and give me a place where I can start. I have forked and cloned the repo and have the Debugger up and running. Just need help in start off.

Thanks!
